### ISSUE TITLE
THORChain - Enforce MinOut in borrows

### ISSUE THREAD

**Issue by gomesalexandre** [shapeshift/web#6954]
### Overview

Currently, we have no `MinOut` in borrows, i.e empty limit component. That's actually as returned by us by THOR, see their example memo in the Swagger docs: `$+:ETH.ETH:0x1c7b17362c84287bd1184447e6dfeaf920c31bbe`.

Skip to the bottom for the tl;dr.

Now, in theory, a limit should be able to be added, as per all THOR "deposit" memos: https://dev.thorchain.org/concepts/memos.html#open-loan
> :MINOUT | Minimum debt amount, else a refund. Similar to :LIM. | Optional. 1e8 format.

So all we need to do is add the returned `expected_amount_out` from the quote as a limit and call it a day, right?
...Right?
Wrong. In reality, things get much, much weirder.

There is an undocumented behavior of THOR defaulting to internal streaming swaps with an absence of limit component.
The `expected_amount_out` we're getting is assuming we use the returned memo (which doesn't contain a limit) as-is, and is more for informational purposes than anything else.

Trying to add said `expected_amount_out` in our memo will result in failures: https://viewblock.io/thorchain/tx/d76ed393598e60ec6d1aef287023c950223fa5db59f46250b7c5754cd9ea3644

The reason is this makes the loan a non-streaming loan, with slightly worsened out amount, and so the initial `expected_amount_out` returned to us doesn't cut it anymore.

There is also another component of confusion here: is `fees.slippage_bps` ever our guy as far as fees our concerned, i.e if circling back to the original issue which prompted us to enforce `MinOut` here (a user getting a very, very bad rate with a specific token as a borrow asset on Avalanche), would `fees.slippage_bps` have caught that?

If so, we may not need further work here at all and should be able to keep relying on the empty limit component, benefiting from streaming swaps, now that `<WarningAcknwledgement />` is implemented for borrows:

https://github.com/shapeshift/web/blob/537f6b872a5a871a17fb393ed810ab61d13b4616/src/pages/Lending/Pool/components/Borrow/BorrowInput.tsx#L492-L497

Sure, we could add a limit in the memo for extra safety, but it wouldn't do much, as we'd really be adding slippage of the streaming loan `expected_amount_out` (already all fees deducted, including slippage) accounting for the worsening of quote and would really be guarding against things being *even worse* than originally quoted at execution time, not against high slippage. 

tl;dr what are we solving here, is this user safety at execution time? If we *can't* make users safer at execution time (and probably shouldn't, as they acknowledged high slippage already), and we now have UX/UI to properly warn them, which is how they ended up in such a situation to begin with, is there still value in adding a limit that doesn't do much in terms of protection? 

### References and additional details

- `<WarningAcknowledgement />` implementation for loan opens:

https://github.com/shapeshift/web/blob/537f6b872a5a871a17fb393ed810ab61d13b4616/src/pages/Lending/Pool/components/Borrow/BorrowInput.tsx#L492-L497

- `assertAndProcessMemo` for loan opens in develop currently:

https://github.com/shapeshift/web/blob/537f6b872a5a871a17fb393ed810ab61d13b4616/src/lib/utils/thorchain/memo/assertAndProcessMemo.ts#L162-L163

### Acceptance Criteria

Either:

- We are able to implement reliable MinOut which is actually protecting users
- We confirm that `fees.slippage_bps` reflects high-slippage situations, close this as wontfix, and add an extra comment to all the currently commented out bits to clearly mention we should *never* uncomment it as this will result in refunds

**Comment by 0xApotheosis** [comment]
Some terminology that might be useful to help us disambiguate some concepts here:

- **Market impact** - the impact to the underlying pools caused by the user performing a large trade. We know this amount at quote time. We've also called this "price impact" in the swapper. We already "protect" against this by warning the user before proceeding with their quote. Note, we incorrectly call this slippage in the UI.
- **Slippage** - the difference between the expected price of a trade and the actual price at which the trade is executed. This is the execution time component mentioned above, which our users are exposed to, but _should_ be protected against via the THORChain streaming magic.
- **Slippage tolerance** - the amount of execution time slippage that the user will tolerate - as you've noted above, we have no way to implement this without effectively giving the user a worse (non-streaming) rate.

**Comment by gomesalexandre** [comment]
> Some terminology that might be useful to help us disambiguate some concepts here:
> 
> * **Market impact** - the impact to the underlying pools caused by the user performing a large trade. We know this amount at quote time. We've also called this "price impact" in the swapper. We already "protect" against this by warning the user before proceeding with their quote. Note, we incorrectly call this slippage in the UI.
> * **Slippage** - the difference between the expected price of a trade and the actual price at which the trade is executed. This is the execution time component mentioned above, which our users are exposed to, but _should_ be protected against via the THORChain streaming magic.
> * **Slippage tolerance** - the amount of execution time slippage that the user will tolerate - as you've noted above, we have no way to implement this without effectively giving the user a worse (non-streaming) rate.

Great disambiguation @0xApotheosis!
Correct on all 3 points, although with one little disambiguation in the disambiguation: 

-**Market impact** may be "protected" (in form of a UI big red warning) against simply as a side-effect of *Slippage*, in reality, all we're doing is leveraging slippage_bps from the quote, which has its own ramifications and may not necessarily warn against price impact (which may be unpredictable by THOR if there is high volatility at the time of the trade). We're also doing quite basic of a check of slippage (denominated in the borrow asset) against collateral amount (denominated in the collateral asset) which doesn't account for collateralization ratio.

Happy to go through this next sync, but there are a few ways we can improve the current ack:

- Either go the simpler and more accurate way of reflecting the slippage_bps as-is (vs. slippage denominated borrow asset USD / collateral amount USD currently), avoiding collateralization ratio ramifications altogether, and assuming everything goes well
- Or go the slightly more complex, but even more accurate way of keeping the current logic, but instead of simply checking borrow side slippage USD / collateral USD, check USD slippage against what *would* be the feeless borrowed amount according to the CR

**Comment by gomesalexandre** [comment]
Next steps here: check the slippage detection math  is correct and honors CR, and don't try and enforce MinOut for borrows

**Comment by twblack88** [comment]
lending go sunset. :sad:
